packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -90,6 +127,60 @@ private TaskResponseHandler getModelCardResponseHandler() {
 		return handler;
 	}
 
+	private TaskResponseHandler configureModelResponseHandler() {
+		final TaskResponseHandler handler = new TaskResponseHandler();
+		handler.onSuccess((TaskResponse resp) -> {
+			try {
+				final String serializedString = objectMapper.writeValueAsString(resp.getAdditionalProperties());
+				final ConfigureModelProperties props = objectMapper.readValue(serializedString, ConfigureModelProperties.class);
+
+				final Model model = modelService.getAsset(props.getModelId())
+						.orElseThrow();
+				final ConfigureModelResponse configurations = objectMapper.readValue(resp.getOutput(), ConfigureModelResponse.class);
+
+				configurations.response.get("conditions").forEach((condition) -> {
+
+
+					final Model modelCopy = model;
+					final List<ModelParameter> modelParameters = modelCopy.getSemantics().getOde().getParameters();
+					modelParameters.forEach((parameter) -> {
+						JsonNode conditionParameters = condition.get("parameters");
+						conditionParameters.forEach((conditionParameter) -> {
+							if (parameter.getId().equals(conditionParameter.get("id").asText())) {
+								parameter.setValue(conditionParameter.get("value").doubleValue());
+							}
+						});
+					});
+					modelCopy.getSemantics().getOde().setParameters(modelParameters);
+
+					final ModelConfiguration configuration = new ModelConfiguration();
```

Comment:
Could you add comment on each step for readability

Reply:
done